modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/ExceptionUtils.cs

```diff
@@ -135,5 +135,122 @@ public static void LogUnhandledException(Exception e)
                 OnExceptionLoggerException(unexpected, e);
             }
         }
+
+        [Conditional("DEBUG")]
+        public unsafe static void DebugCheckCallError(string method, IntPtr instance, godot_variant** args, int argCount, godot_variant_call_error error)
+        {
+            if (error.Error != godot_variant_call_error_error.GODOT_CALL_ERROR_CALL_OK)
+            {
+                using godot_variant instanceVariant = VariantUtils.CreateFromGodotObjectPtr(instance);
+                string where = GetCallErrorWhere(method, &instanceVariant, args, argCount);
+                string errorText = GetCallErrorMessage(error, where, args);
+                GD.PushError(errorText);
+            }
+        }
+
+        [Conditional("DEBUG")]
+        public unsafe static void DebugCheckCallError(in godot_callable callable, godot_variant** args, int argCount, godot_variant_call_error error)
+        {
+            if (error.Error != godot_variant_call_error_error.GODOT_CALL_ERROR_CALL_OK)
+            {
+                using godot_variant callableVariant = VariantUtils.CreateFromCallableTakingOwnershipOfDisposableValue(callable);
+                string where = GetCallErrorWhere("Call", &callableVariant, args, argCount);
+                string errorText = GetCallErrorMessage(error, where, args);
+                GD.PushError(errorText);
+            }
+        }
+
+        private unsafe static string GetCallErrorWhere(string method, godot_variant* instance, godot_variant** args, int argCount)
+        {
+            string methodstr = method;
+            string basestr = GetVariantTypeName(instance);
+
+            switch (method)
+            {
+                case "Call":
+                    if (argCount >= 1 && instance->Type != Variant.Type.Callable)
+                    {
+                        methodstr = $"{VariantUtils.ConvertToString(*args[0])} (via call)";
+                    }
+                    else
+                    {
+                        methodstr = $"{VariantUtils.ConvertToString(*instance)} (Callable)";
+                    }
+                    break;
+                case "CallRecursive" when basestr == "Godot.TreeItem":
+                    if (argCount >= 1)
+                    {
+                        methodstr = $"{VariantUtils.ConvertToString(*args[0])} (via TreeItem.CallRecursive)";
+                    }
+                    break;
+            }
+
+            methodstr = $"function '{methodstr}'";
```

Comment:
(it is the same in gdscript but) IMO it is kinda weird that the `(via call)` ends up inside the single quotes, I would only keep the method name in the quotes and then have the parenthesized part after.

Reply:
I agree, we don't have to use the exact same messages that GDScript uses but I thought it would be a good starting point. Do you have any opinions on the other messages?

Reply:
TBH it's jsut this little detail that annoyed me, but if we (slightly) change these errors I would simplify this whole method a bit:
```cs
        private unsafe static string GetCallErrorWhere(string method, godot_variant* instance, godot_variant** args, int argCount)
        {
            return method == "Call" || method == "CallRecursive"
                ? instance->Type == Variant.Type.Callable
                    ? $"callable '{VariantUtils.ConvertToString(*instance)}'"
                    : $"function '{VariantUtils.ConvertToString(*args[0])}' in base '{GetVariantTypeName(instance)}' via {method}"
                : $"function '{method}' in base '{GetVariantTypeName(instance)}'"
        }
```
ie I would put the via at the end and have `callable` as a prefix instead of `function`

Reply:
I'm not sure how I feel about nested ternary operators. Also, not sure if we should remove the check for `TreeItem` since then we assume every `CallRecursive` method will have the method name as its first parameter and that may not always be the case.

Reply:
just a suggestion, mostly about the message format coupled with an observation that the method can be simplified a lot, except that it didn't exactly work as well as I had guessed. Using proper if statements and adjusting the conditions is probably best.

Reply:
I tried simplifying it a bit and changed the method string to be a `StringName` so I can use the cached names in `MethodName`.